merge/owners.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package merge
+
+import "sigs.k8s.io/structured-merge-diff/fieldpath"
+
+// APIVersion describes the version of an object or of a fieldset.
+type APIVersion string
+
+// VersionedSet associates a version to a set.
+type VersionedSet struct {
+	*fieldpath.Set
+	APIVersion APIVersion
+}
+
+// Owners is a map from owner to VersionedSet (what they own in
+// what version).
+type Owners map[string]*VersionedSet
+
+// Difference returns a symmetric difference between two Owners.  If the
+// version is different for a user, then the entire rhs set is returned
+// with rhs version. Empty sets are not inserted at all, so a zero-length
+// check will verify that the difference is empty
```

Comment:
I don't understand this last sentence. Does it mean "users which own no fields will be omitted from the return value"?